### PR TITLE
Configure firewall rules using nftables

### LINKED ISSUE #1112: Add support of Ubuntu 22

### PR DESCRIPTION
**Pull request by badrogger**
**Changes**
- Use nftables instead of iptables to configure rules. 

**Testing**
- Tested on local network.
- Added unit tests. 
 
**Performance**
- Changes should not affect performance of the system.

### CODE REVIEW

**Review by DmytroNazarenko — DISMISSED** [review]

File: core/schains/firewall/firewall_manager.py [hunk 1]
```diff
@@ -83,8 +89,17 @@ def remove_rules(self, rules: Iterable[SChainRule]) -> None:
...
 
 class IptablesSChainFirewallManager(SChainFirewallManager):
     def create_host_controller(self) -> IptablesController:
         return IptablesController()
+
+
+class NFTSchainFirewallManager(SChainFirewallManager):
+    def create_host_controller(self) -> NFTablesController:
+        nc_controller = NFTablesController(chain=self.name)
+        nc_controller.create_table()
```
Comment (dmytrotkk) [inline core/schains/firewall/firewall_manager.py:103]: let's rename

File: core/schains/firewall/nftables.py [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+import logging
+import multiprocessing
+import os
+from typing import Iterable, TypeVar
+
+from core.schains.firewall.types import IHostFirewallController, SChainRule
+
+from tools.configs import NFT_CHAIN_BASE_PATH
+
+T = TypeVar('T')
```
Comment (dmytrotkk) [inline core/schains/firewall/nftables.py]: looks like `T = TypeVar('T')` is unused

File: core/schains/firewall/nftables.py [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+                            'table': self.table,
+                            'chain': self.chain,
+                            'expr': expr,
+                        }
+                    }
+                }
+            ]
+        )
+
+        rc, output, error = self.run_json_cmd(json_cmd)
```
Comment (dmytrotkk) [inline core/schains/firewall/nftables.py]: output is unused, can be replaced with `_`

File: core/schains/firewall/nftables.py [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+        if rc != 0:
+            raise NFTablesCmdFailedError(f'Failed to add allow rule: {error}')
+
+    @classmethod
```
Comment (dmytrotkk) [inline core/schains/firewall/nftables.py]: does it make sense to group all class methods and move them to the beginning or the end of the class?

**Review by dmytrotkk — CHANGES_REQUESTED** [review]

**Review by dmytrotkk — APPROVED** [review]

**Review by DmytroNazarenko — APPROVED** [review]

**Pull request merged**